subprojects/packagefiles/zlib-ng/meson.build

```diff
@@ -1396,7 +1396,11 @@ zlib_ng_dep = declare_dependency(
   link_with: zlib_ng,
   include_directories: include_directories('.'),
 )
-meson.override_dependency('zlib-ng', zlib_ng_dep)
+if not get_option('zlib-compat')
+  meson.override_dependency('zlib-ng', zlib_ng_dep)
```

Comment:
We already define the pkgconfig export name as `'zlib' + suffix` and the override_dependency must match the pkgconfig export name.

This PR does do that, so :+1:, but for style reasons please use the same `+ suffix` pattern here rather than a new option check.